src/renderer/gl_engine/tvgGlRenderer.cpp

```diff
@@ -627,25 +633,25 @@ void GlRenderer::prepareCmpTask(GlRenderTask* task)
     // left top point
     vertices.push(left);
     vertices.push(top);
-    vertices.push(1.f);
+    vertices.push(opacity);
     vertices.push(0.f);
     vertices.push(1.f);
     // left bottom point
     vertices.push(left);
     vertices.push(bottom);
-    vertices.push(1.f);
+    vertices.push(opacity);
     vertices.push(0.f);
     vertices.push(0.f);
     // right top point
     vertices.push(right);
     vertices.push(top);
-    vertices.push(1.f);
+    vertices.push(opacity);
     vertices.push(1.f);
     vertices.push(1.f);
     // right bottom point
     vertices.push(right);
     vertices.push(bottom);
-    vertices.push(1.f);
+    vertices.push(opacity);
     vertices.push(1.f);
     vertices.push(0.f);
 
```

Comment:
Should we pass additional opacity value via vertex data? As I can see there is general value for the mesh, and we are just doubling this information

Reply:
This is just for convenience as it's just a single float value and I don't want to create a separate props and uniform buffer to store and pass it around. :)